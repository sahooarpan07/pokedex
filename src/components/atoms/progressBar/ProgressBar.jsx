import React from 'react';
import {View, Text} from 'react-native';
import * as Progress from 'react-native-progress';
import {styles} from './ProgressBar.styles';

const ProgressBar = ({label, value}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <View style={styles.progressBarContainer}>
        <Progress.Bar
          progress={value}
          height={12}
          width={200}
          color="#180345"
          style={styles.progressBar}
        />
      </View>
      <Text style={styles.value}>{Math.round(value * 100)}</Text>
    </View>
  );
};

export default ProgressBar;
