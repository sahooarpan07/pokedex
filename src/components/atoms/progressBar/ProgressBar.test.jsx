/**
 * @format
 */

import 'react-native';
import React from 'react';
import ProgressBar from './ProgressBar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render Progress Bar', () => {
  const tree = renderer
    .create(<ProgressBar label={'HP'} value={68} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
