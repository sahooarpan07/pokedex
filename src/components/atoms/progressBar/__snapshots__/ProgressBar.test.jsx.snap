// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Progress Bar 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      {
        "color": "#180345",
        "fontSize": 18,
        "fontWeight": "400",
        "textAlign": "center",
      }
    }
  >
    HP
  </Text>
  <View
    style={
      {
        "margin": 8,
      }
    }
  >
    <View
      animated={true}
      animationConfig={
        {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={false}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={68}
      style={
        [
          {
            "backgroundColor": undefined,
            "borderColor": "#180345",
            "borderRadius": 4,
            "borderWidth": 1,
            "overflow": "hidden",
            "width": 200,
          },
          {
            "position": "relative",
          },
        ]
      }
      useNativeDriver={false}
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#180345",
            "height": 12,
            "transform": [
              {
                "translateX": -7.105427357601002e-15,
              },
              {
                "translateX": 0,
              },
              {
                "scaleX": 1,
              },
            ],
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontWeight": "bold",
        "left": 200,
        "position": "absolute",
        "top": 4,
      }
    }
  >
    6800
  </Text>
</View>
`;
